src/Options/Factory.hpp

```diff
@@ -51,16 +51,36 @@ struct print_derived {
   }
 };
 
-template <typename BaseClass>
+template <typename CreatableClasses>
 std::string help_derived() noexcept {
   return "Known Ids:\n" +
-         tmpl::for_each<typename BaseClass::creatable_classes>(
+         tmpl::for_each<CreatableClasses>(
              Factory_detail::print_derived{})
              .value;
 }
 
+// This is for handling legacy code that still uses creatable_classes.
+// It should be inlined once everything is converted.
+template <typename BaseClass, typename Metavariables, typename = std::void_t<>>
+struct get_creatable_classes {
+  using type = tmpl::at<typename Metavariables::factory_classes, BaseClass>;
+};
+
+template <typename BaseClass, typename Metavariables>
+struct get_creatable_classes<
+  BaseClass, Metavariables,
+  std::void_t<typename BaseClass::creatable_classes>> {
+  using type = typename BaseClass::creatable_classes;
+};
+
 template <typename BaseClass, typename Metavariables>
 std::unique_ptr<BaseClass> create(const Option& options) {
+  using creatable_classes =
+      typename get_creatable_classes<BaseClass, Metavariables>::type;
+  static_assert(not std::is_same_v<creatable_classes, tmpl::no_such_type_>,
```

Comment:
I wasn't aware of `tmpl::no_such_type_`, this is a good way to print useful compiler errors 👍